### ISSUE TITLE
Placing order in lesson 2 will not work when following guide.

### ISSUE THREAD

**Issue by 17steen** [dotnet-presentations/blazor-workshop#339]
[save point 2 Address.cs](https://github.com/dotnet-presentations/blazor-workshop/blob/main/save-points/02-customize-a-pizza/BlazingPizza.Shared/Address.cs), in this file, required fields are defined. Those do not get filled up at any point by the Client, but the server will expect them, when the order is placed, to be filled in.
This however never gets mentioned, and the tutorial goes on to lesson 3, which will then never work because the orders do not get added to the database.
It gets very confusing as you then think to start from the following save point, and now it magically works. This is the result of those constraints to now be commented, without it being shown in the tutorial. This lead the learner to have a considerably lengthy headache not understanding what they did wrong. [save point 3 Address.cs](https://github.com/dotnet-presentations/blazor-workshop/blob/main/save-points/03-show-order-status/BlazingPizza.Shared/Address.cs)

**Comment by catalinp86** [comment]
Same issue. Tutorial says that if I click on Track link button next to the order it will go to my orders/<id>

Only problem is that I never get to have any orders as they never get to the back-end. It's very confusing.

**Comment by ylr-research** [comment]
@catalinp86 I haven't seen this issue. Mine works just fine. Could you share your repo?

**Comment by ylr-research** [comment]
@17steen true that save-point sections -even 1st- have the required attributes. That should be fixed so it matches the docs. However, I would encourage to follow the tutorial while save-points is fixed.

**Comment by catalinp86** [comment]
I will try again, a 3rd time, this time on another machine and see if I reproduce it again.  This time, I'll clone it on my machine, take the state at end of the session 2, place it in a public github repo and add the first steps in session 3. If the problem is still there I will push the changes and leave it as is.

**Comment by catalinp86** [comment]
> @catalinp86 I haven't seen this issue. Mine works just fine. Could you share your repo?

@ylr-research I have repeated once again, the same process:

1. Taken the source code from 02-customize-a-pizza
2. Added the MyOrders part from 03 Show Order Status
3. Run the application once I've added the Rendering of the list of orders
4. Observed that I can place an order, but no orders are displayed in the MyOrders page 

It adds up, because after submitting the pizza order, when I enter the MyOrders page, when the HttpClient attempts to get the list of Pizza orders it comes back with an empty list. That's abnormal.

**Comment by DjinnYcs** [comment]
I met same problem, and somehow magically it solved by itself. 
Firstly, I commented [Address.cs](https://github.com/dotnet-presentations/blazor-workshop/blob/main/save-points/03-show-order-status/BlazingPizza.Shared/Address.cs) as in 3rd save point and began to compare code lines of 3rd and then 2nd save points in debug mode. Breakpoints was chosen to check **orders** variables (as @catalinp86 mentioned), and I didn't find any obvious differences, BUT in debug mode the 2nd save point surprisely start working as expected in tutorial. It's worth noting, that I launched debugging during whole day, but issue was resolved only after put breakpoints and checking lines step by step. Strange bug.

**Comment by catalinp86** [comment]
I'll test that out @rzmz

**Comment by neintoes** [comment]
Thanks very much for this @17steen and @rzmz. I experienced the same issue as @catalinp86 . Commenting out model validation attributes didn't work initially but as you said re-creating the database got it all working again. Much appreciated.

**Comment by csharpfritz** [comment]
Yes, I think we need to roll-back the initial requirements attributes in Address.cs as a recommended fix to this

**Comment by Gobbenobber** [comment]
Just wanted to add I only created a new PR because I couldn't figure out how to contribute to the existing one that's 27 days old and is missing changes to save-points 01 and 02.
